grid_map_core/src/GridMap.cpp

```diff
@@ -717,5 +762,62 @@ void GridMap::resize(const Index& size)
   }
 }
 
+void GridMap::conservativeResize(const Index& size, const Direction direction, float value)
+{
+  int row_delta = size(0) - size_(0);
+  int col_delta = size(1) - size_(1);
+
+  Eigen::MatrixXf like = Eigen::MatrixXf::Constant(size(0), size(1), value);
+  for (auto& data : data_)
+  {
+    data.second.conservativeResizeLike(like);
```

Comment:
Plain curiosity,  
why are you using `conservativeResizeLike` rather than `conservativeResize` ??  
The function `conservativeResize` allocates a temporary matrix under the hood, so I would imagine that `conservativeResizeLike` does the same. Thus isn't the temporary matrix `like` a waste of memory ?

Reply:
`conservativeResize` doesn't let you initialize the values of new elements, `conservativeResizeLike` does. This is important when you want to be able to guarantee that every new grid cell is initalized to 0.0, for instance.

In principle I could use `conservativeResize`, loop through all new cells and manually initialize them, but I suspect that would be less efficient.

Reply:
Ok I see your point.
Still I think a plain `conservativeResize` followed by manually setting the uninitialized values would be better as you don't really need allocating matrix `like`, especially as the matrix grows large.  
Rather than looping over uninitialized values you can use the `block` functions, something along the lines :
```cpp
data.second.conservativeResize(size(0), size(1));
// Assuming that resizing to larger dimension appends 
// rows at the bottom and cols at the right side.
// Set value for appended bottom rows
data.second.block(size_(0), 0, row_delta, size(1)).setConstant(value); 
// Set value for appended top-left block (remaining unintialized)
data.second.topRightCorner(size(0), col_delta).setConstant(value);
```
I haven't checked that the indices are actually correct ^^, but that's the general idea.

Reply:
Is this feature working? What is it stopping from going to production?

Reply:
Just waiting for an approving review. We have been using this feature extensively in production at our [company](https://github.com/Maidbot/).

Reply:
this is very nice feature and exactly what I needed! A merge would be good.

Reply:
+1 for useful feature and would be good to have merged

Reply:
Yes, will import it.